zebrad/src/components/sync/tests/timing.rs

```diff
@@ -75,11 +73,11 @@ fn request_genesis_is_rate_limited() {
     // The number of calls to `request_genesis()` we are going to be testing for
     const RETRIES_TO_RUN: u8 = 3;
 
-    // create some counters that will be updated inside async blocks
-    let peer_requests_counter = Arc::new(AtomicU8::new(0));
-    let peer_requests_counter_in_service = Arc::clone(&peer_requests_counter);
-    let state_requests_counter = Arc::new(AtomicU8::new(0));
-    let state_requests_counter_in_service = Arc::clone(&state_requests_counter);
+    // create some watchers that will be updated inside async blocks
+    let mut peer_requests_counter = 0;
+    let mut state_requests_counter = 0;
+    let (peer_requests_sender, peer_requests_receiver) = watch::channel(peer_requests_counter);
+    let (state_requests_sender, state_requests_receiver) = watch::channel(state_requests_counter);
```

Comment:
The error seems to be caused by `ChainSync` needing to clone the service to create multiple instances of it. To allow that, I think one way would be to simply wrap the senders with `Arc`s so that the `watch::Sender` is shared between all instances:

```suggestion
    let (peer_requests_sender, peer_requests_receiver) = watch::channel(peer_requests_counter);
    let (state_requests_sender, state_requests_receiver) = watch::channel(state_requests_counter);
    let peer_requests_sender = Arc::new(peer_requests_sender);
    let state_requests_sender = Arc::new(state_requests_sender);
```